From 533eccfd2b6ba10183b4dd7c7c901cc23bfef0bc Mon Sep 17 00:00:00 2001
From: ngotienloc <ngotienlocf@gmail.com>
Date: Wed, 19 Nov 2025 10:56:12 +0700
Subject: [PATCH] MLFQ: Final structural integration

---
 include/linux/sched.h |  2 ++
 kernel/sched/Makefile | 16 ++++++++++++++++
 2 files changed, 18 insertions(+)

diff --git a/include/linux/sched.h b/include/linux/sched.h
index 0d1d70ade..59b66c608 100644
--- a/include/linux/sched.h
+++ b/include/linux/sched.h
@@ -827,6 +827,8 @@ struct task_struct {
 	 */
 	int				recent_used_cpu;
 	int				wake_cpu;
+	#ifdef CONFIG_SCHED_MLFQ
+    struct mlfq_task_info   mlfq;
 #endif
 	int				on_rq;
 
diff --git a/kernel/sched/Makefile b/kernel/sched/Makefile
index 976092b7b..98bdda0f1 100644
--- a/kernel/sched/Makefile
+++ b/kernel/sched/Makefile
@@ -32,3 +32,19 @@ obj-y += core.o
 obj-y += fair.o
 obj-y += build_policy.o
 obj-y += build_utility.o
+
+# File: ~/pios/kernel/linux/kernel/sched/Makefile
+
+obj-$(CONFIG_SCHED_DEADLINE) += deadline.o
+obj-$(CONFIG_SCHED_AUTOGROUP) += auto_group.o
+
+ifdef CONFIG_SCHED_MLFQ
+  # Biên dịch code MLFQ nếu CONFIG_SCHED_MLFQ được bật
+  obj-y += mlfq.o
+endif
+
+ifdef CONFIG_SCHED_DEBUG
+  CFLAGS_core.o += $(call tune_cc,CHECK_ENDIAN)
+endif
+
+# ... các dòng khác ...
\ No newline at end of file
-- 
2.43.0

